test/utils.toNumber.js

```diff
@@ -2,14 +2,22 @@ var assert = require('assert');
 var utils = require('../packages/web3-utils');
 
 describe('lib/utils/utils', function () {
-    describe('hexToNumberString', function () {
+    describe('hexToNumber', function () {
         it('should return the correct value', function () {
 
-            assert.equal(utils.hexToNumberString("0x3e8"), 1000);
-            assert.equal(utils.hexToNumberString('0x1f0fe294a36'), 2134567897654);
+            assert.equal(utils.hexToNumber("0x3e8"), 1000);
+            assert.equal(utils.hexToNumber('0x1f0fe294a36'), 2134567897654);
             // allow compatiblity
-            assert.equal(utils.hexToNumberString(100000), 100000);
```

Comment:
```javascript
// allow compatibility
```
... is an important warning. Maybe these tests should be unchanged and we should clarify what intended hexToNumber behavior is here and in the [docs](https://web3js.readthedocs.io/en/v1.2.1/web3-utils.html#hextonumber).

Reply:
The function ``hexToNumber`` should do what it's saying. I would enforce to have all passed hex values prefixed with ``0x`` if the given value isn't prefixed with ``0x`` should it throw an error.

Reply:
Yes it should. Current behavior is a little unusual.